pennylane/transforms/commutation_dag.py

```diff
@@ -0,0 +1,1049 @@
+                        qml.Rot(*operation1.data, wires=target_wires_1).get_matrix(),
+                    ),
+                )
+            )
+
+        if operation2.name == "CRot":
+            if not intersection(target_wires_2, operation1.wires):
+                return bool(commutation_map[operation1.name][position["ctrl"]])
+            return np.all(
+                np.allclose(
+                    np.matmul(
+                        qml.Rot(*operation2.data, wires=target_wires_2).get_matrix(),
+                        operation1.get_matrix(),
+                    ),
+                    np.matmul(
+                        operation1.get_matrix(),
+                        qml.Rot(*operation2.data, wires=target_wires_2).get_matrix(),
+                    ),
+                )
+            )
+
+        return np.all(
+            np.allclose(
+                np.matmul(
+                    operation1.get_matrix(),
+                    operation2.get_matrix(),
+                ),
+                np.matmul(
+                    operation2.get_matrix(),
+                    operation1.get_matrix(),
+                ),
+            )
+        )
+    return None
+
+
+unsupported_operations = [
+    "PauliRot",
+    "QubitDensityMatrix",
+    "CVNeuralNetLayers",
+    "ApproxTimeEvolution",
+    "ArbitraryUnitary",
+    "CommutingEvolution",
+    "DisplacementEmbedding",
+    "SqueezingEmbedding",
+]
+non_commuting_operations = [
+    "ArbitraryStatePreparation",
+    "BasisStatePreparation",
+    "MottonenStatePreparation",
+    "QubitCarry",
+    "QubitSum",
+    "SingleExcitation",
+    "SingleExcitationMinus",
+    "SingleExcitationPlus",
+    "DoubleExcitation",
+    "DoubleExcitationPlus",
+    "DoubleExcitationMinus",
+    "BasicEntanglerLayers",
+    "GateFabric",
+    "ParticleConservingU1",
+    "ParticleConservingU2",
+    "RandomLayers",
+    "SimplifiedTwoDesign",
+    "StronglyEntanglingLayers",
+    "AllSinglesDoubles",
+    "FermionicDoubleExcitation",
+    "FermionicSingleExcitation",
+    "Grover",
+    "kUpCCGSD",
+    "Permute",
+    "QFT",
+    "QuantumMonteCarlo",
+    "QuantumPhaseEstimation",
+    "UCCSD",
+    "MPS",
+    "TTN",
+    "AmplitudeEmbedding",
+    "AngleEmbedding",
+    "BasisEmbedding",
+    "IQPEmbedding",
+    "QAOAEmbedding",
+]
+
+
+def is_commuting(operation1, operation2):
+    r"""Check if two operations are commuting. A lookup table is used to check the commutation between the
+    controlled, targeted part of operation 1 with the controlled, targeted part of operation 2. It supports
+    most PennyLane operations that are not CV operations. Unsupported operation are the following:
+
+    :class:`qml.PauliRot`, :class:`qml.QubitDensityMatrix`, :class:`qml.CVNeuralNetLayers`,
+    :class:`qml.ApproxTimeEvolution`, :class:`qml.ArbitraryUnitary`, :class:`qml.CommutingEvolution`,
+    :class:`qml.DisplacementEmbedding` and :class:`qml.SqueezingEmbedding`.
```

Comment:
```suggestion
    :class:`~.PauliRot`, :class:`~.QubitDensityMatrix`, :class:`~.CVNeuralNetLayers`,
    :class:`~.ApproxTimeEvolution`, :class:`~.ArbitraryUnitary`, :class:`~.CommutingEvolution`,
    :class:`~.DisplacementEmbedding` and :class:`~.SqueezingEmbedding`.
```

Since `qml` is just a convention, Sphinx will unfortunately not be able to find and resolve these links (see https://pennylane--1712.org.readthedocs.build/en/1712/code/api/pennylane.is_commuting.html).

Instead, preceding the name with `.` instructs sphinx to attempt to automatically find the first object that corresponds (e.g., `:class:.ObjectName`). This will render as `pennylane.path.to.ObjectName`. Including the `~` then instructs Sphinx to only render it as `ObjectName` 🙂

Reply:
Thank you, you are right! I didn't pay attention.